File: 0201-BitwiseANDOfNumbersRange/BitwiseANDOfNumbersRange.go
package bitwiseandofnumbersrange

func rangeBitwiseAnd(left int, right int) int {
	bitShifted := 0
	for left < right {
		left >>= 1
		right >>= 1
		bitShifted++
	}

	ans := left << bitShifted
	return ans
}

File: 0201-BitwiseANDOfNumbersRange/BitwiseANDOfNumbersRange.java

class BitwiseANDOfNumbersRange {
    class Solution {
        public int rangeBitwiseAnd(int left, int right) {
            int bitShifted = 0;
            while(left < right){
                left >>= 1;
                right >>= 1;
                bitShifted++;
            }
            
            
            return left << bitShifted;
        }
    }
}
